go/registry/api/api.go

```diff
@@ -426,14 +426,14 @@ func VerifyRegisterNodeArgs( // nolint: gocyclo
 		expectedSigners = append(expectedSigners, entity.ID)
 	}
 
-	// Checking the expiration only makes sense if this routine is
-	// validating the genesis document, since node descriptors being
-	// expired is presumably ok at runtime(?).
-	if isGenesis && n.Expiration <= uint64(epoch) {
-		logger.Error("RegisterNode: expired node in genesis",
+	// Registrations that are expired are allowed in the genesis document
+	// as expired node registrations are preserved through the debonding
+	// period of an entity.
+	if !isGenesis && n.Expiration <= uint64(epoch) {
```

Comment:
mind also deduplicating this with: https://github.com/oasislabs/oasis-core/blob/master/go/consensus/tendermint/apps/registry/transactions.go#L242-L257 (more ore less removing the check there, as that code path already calls `VerifyRegisterNodeArgs`)

Reply:
(or at minimum the check there needs to be fixed as well to only check for non-genesis)

Reply:
No, I can't deduplicate this, because the supplementary sanity checker calls it, and it will force crash the node when it encounters descriptors in the debonding period.